core/src/main/java/hudson/tasks/ArtifactArchiver.java

```diff
@@ -231,18 +243,45 @@ public void perform(Run<?,?> build, FilePath ws, Launcher launcher, TaskListener
         try {
             String artifacts = build.getEnvironment(listener).expand(this.artifacts);
 
-            Map<String,String> files = ws.act(new ListFiles(artifacts, excludes, defaultExcludes, caseSensitive));
+            FilePath dir = ws;
+            if (this.basePath != null && this.basePath.length() > 0) {
```

Comment:
That's really required only when you have overlapped variable. Enforces to start searching whether `basePath` additionally exist and handled.